src/encoded/commands/generate_ontology.py

```diff
@@ -746,38 +757,41 @@ def main():
 
     # fourfront connection
     connection = connect2server(args.env, args.key)
+    print("Pre-processing")
     ontologies = get_ontologies(connection, args.ontologies)
     for i, o in enumerate(ontologies):
-        if o['ontology_name'].startswith('4DN'):
+        if o['ontology_name'].startswith('4DN') or o['ontology_name'].startswith('CGAP'):
             ontologies.pop(i)
     slim_terms = get_slim_terms(connection)
-    from_ontologies = None
-    if (args.ontologies != 'all'):
-        from_ontologies = ontologies
-    db_terms = get_existing_ontology_terms(connection, from_ontologies)
+    db_terms = get_existing_ontology_terms(connection)
     terms = {}
 
     for ontology in ontologies:
         print('Processing: ', ontology['ontology_name'])
         if ontology.get('download_url', None) is not None:
+            # want only simple processing for HP
+            simple = False
+            if ontology.get('ontology_prefix') in ['HP', 'EFO'] and not args.imports:
+                simple = True
             # get all the terms for an ontology
-            terms = download_and_process_owl(ontology, connection, terms)
+            terms = download_and_process_owl(ontology, connection, terms, simple)
 
     # at this point we've processed the rdf of all the ontologies
     if terms:
+        print("Post-processing")
         terms = add_slim_terms(terms, slim_terms)
         terms = remove_obsoletes_and_unnamed(terms)
-        terms = verify_and_update_ontology(terms, ontologies)
+        terms = verify_and_update_ontology(terms, ontologies, db_terms)
         filter_unchanged = True
         if args.full:
             filter_unchanged = False
-        partitioned_terms = id_post_and_patch(terms, db_terms, ontologies, filter_unchanged)
-        terms2write = add_uuids_and_combine(partitioned_terms)
+        updates, idmap = id_post_and_patch(terms, db_terms, ontologies, filter_unchanged)
```

Comment:
Do we still want to return the second part of this? It looks like `idmap` is no longer being used outside the function that generates it.

Reply:
You're right this is no longer needed.  Do you want to fix or shall I?